ui/DesignSystem/Component/TrackToggle.svelte

```diff
@@ -6,41 +6,147 @@
 
   export let style = null;
   export let peerCount = null;
+  let hover = false;
+  let active = false;
+  let tracking = false;
+  const enter = () => {
+    hover = true;
+  };
+  const leave = () => {
+    hover = false;
+  };
+  const down = () => {
+    active = true;
+  };
+  const up = () => {
+    active = false;
+    tracking = !tracking;
+  };
 </script>
 
 <style>
   .toggle {
     display: flex;
-    height: 36px;
+    height: 40px;
     align-items: center;
-    border-radius: 2px;
-    border: 1px solid var(--color-tertiary-level-6);
-    background-color: var(--color-background);
+    border-radius: 4px;
+    border: 1px solid var(--color-secondary);
+    background-color: var(--color-secondary);
     color: var(--color-background);
+    cursor: pointer;
+  }
+
+  .toggle.hover {
+    border: 1px solid var(--color-secondary-level-2);
+    background-color: var(--color-secondary-level-2);
+    box-shadow: 0 0 0 1px var(--color-secondary-level-2);
+  }
+  .toggle.active {
+    border: 1px solid var(--color-secondary-level-1);
+    background-color: var(--color-secondary-level-1);
+    box-shadow: 0 0 0 1px var(--color-secondary-level-1);
+  }
+  .toggle.tracking {
+    border: 1px solid var(--color-foreground-level-3);
+    background-color: var(--color-foreground-level-3);
+    color: var(--color-foreground-level-6);
+  }
+  .toggle.tracking.hover {
+    border: 1px solid var(--color-foreground-level-2);
+    background-color: var(--color-foreground-level-2);
+    box-shadow: 0 0 0 1px var(--color-foreground-level-2);
+    color: var(--color-foreground-level-6);
+  }
+  .toggle.tracking.active {
+    border: 1px solid var(--color-foreground-level-2);
+    background-color: var(--color-foreground-level-2);
+    box-shadow: 0 0 0 1px var(--color-foreground-level-2);
+    color: var(--color-foreground-level-4);
   }
 
   .left {
-    background-color: var(--color-tertiary-level-6);
+    background-color: var(--color-secondary);
     display: flex;
-    height: 36px;
+    height: 40px;
     align-items: center;
+    border-top-left-radius: 3px;
+    border-bottom-left-radius: 3px;
+  }
+
+  .left :global(svg) {
```

Comment:
We should be as specific as possible when writing global styles. Otherwise another component with a `.left` class that has an `<svg>` could be affected by this.
```suggestion
  .toggle.left :global(svg) {
```

Reply:
updated